### DISCUSSION TITLE
How to get rid of this error message?

### DISCUSSION THREAD

**Discussion by hw2007** [zed-industries/zed#7637] [Feature Requests]
I am using zed to write a python game, making use of python. Why is this error here, and how do I get rid of it? There isn't anything wrong with my code, this is how the update() function is meant to be used, and this error doesn't show up on other IDEs.
<img width="952" alt="image" src="https://github.com/zed-industries/zed/assets/60082961/d8b4ace8-cf81-4be5-a4c4-627b85868950">

**Comment by ghost** [comment]
The reason you might be seeing this in Zed and not other IDEs is likely the choice of language server (which can have different behaviors). Zed uses [`pyright`](https://github.com/microsoft/pyright). 

Since your `update()` function is overriding the implementation in the `Sprite` class which takes the `*args` and `**kwargs` parameters, I believe this is correct behavior (even though your code will likely run fine without explicitly adding it).

Updating your override to
```python
def update(self, *args, **kwargs):
    ...
```

should get the Language Server to stop giving you that error.

---

Alternatively you can disable that setting in the `pyright` configuration: setting `reportIncompatibleMethodOverride` to `none` should do the trick. See https://microsoft.github.io/pyright/#/configuration. 

This section in Zed docs covers how to configure `pyright`: https://github.com/zed-industries/zed/blob/59bc81d1bcaa634adc44d5ddf51a4b184b725b20/docs/src/languages/python.md#configuration
